### PR TITLE
Rules: Add ability to define tags

### LINKED ISSUE #57: Add ability to define Rule tags
Currently neither JSRule nor RuleBuilder doesn't support adding Tags to the create rule. Maybe there is a reason why it is not available that I'm not aware. Anyway, I can contribute with this addition if necessary.

### PR DESCRIPTION
**Pull request by florian-h05**
@marcinczeczko
Adds the ability to define rule tags in both `JSRule` and `RuleBuilder` syntax.
Tags must be an Array of Strings, in the `RuleBuilder` syntax the tag `js-rule-builder` is always added.

I also updated the README for the tags, completed the Table of Contents and fixed the link to the JS-Joda docs.

Fixes #57.

Signed-off-by: Florian Hotze <florianh_dev@icloud.com>

### CODE REVIEW

**Comment by Confectrician** [comment]
Why did you remove the capitalization in so much (sub)headlines?
Capitalization is a common style practice for docs healdines, even if the terms should/must be used in the code without it.

**Comment by florian-h05** [comment]
@Confectrician 
I thought that the headlines should be the same way in the headline and the table of contents, therefore I removed the capitalization, the namespaces are also written not capitalized.

I can revert back to capitalized headlines, but then I would like to capitalize the entries in the table of contents so that they are the same way.

**Comment by Confectrician** [comment]
From a reading persepective (and that's what the readme aims for) capitalization is the common way for handling the headings, so i would vote for capitalization.

As long as the code examples are not capitalized (only TOC and Headings) this should be fine.
@digitaldan do you have any problems with this?

**Comment by florian-h05** [comment]
@Confectrician 

I have capitalized all headings and I have changed the level of heading for the subheadings of `Actions`, therefore subheadings like `Exec` are not displayed in the ToC anymore.

File: rules/rule-builder.js [hunk 1]
```diff
@@ -89,7 +89,8 @@ class RuleBuilder {
             ruleGroup: optionalRuleGroup,
             execute: function (data) {
                 fnToExecute(data);
-            }
+            },
+            tags: this.tags || ['js-rule-builder']
```
Comment (digitaldan) [inline rules/rule-builder.js:93]: Why set a default tag here?  I would assume if no tags are supplied, we should not add one ?

File: rules/operation-builder.js [hunk 2]
```diff
@@ -17,23 +17,26 @@ class OperationBuilder {
         }
     }
 
-    _then(operation, group, name, description) {
+    _then(operation, group, name, description, tags) {
         this.builder.name = name;
         this.builder.description = description;
+        if (tags) tags.push('js-rule-builder');
```
Comment (digitaldan) [inline rules/operation-builder.js:23]: Why add a default tag here as well?

**Comment by digitaldan** [comment]
Thanks @florian-h05 for the PR!  I have 2 small comments, but otherwise this looks good.

**Review by digitaldan — CHANGES_REQUESTED** [review]
Please see my comments about default tags being added.

File: rules/rule-builder.js (hunk 1 above)
Comment (florian-h05) [reply rules/rule-builder.js:93]: I thought having the `js-rule-builder` tag always added would be helpful, as you would know where a rule comes from in the UI.

**Comment by florian-h05** [comment]
Should I remove the default tags?

**Comment by florian-h05** [comment]
@digitaldan 

I have removed the default tag. Ready for your review.

**Comment by florian-h05** [comment]
@digitaldan ?

**Comment by digitaldan** [comment]
Hi , was not able to work on openHAB at all this week, i try and pull down all PR's and test them before merging.  I'm going to do that later today, thanks.

**Review by digitaldan — APPROVED** [review]
LGTM

**Pull request merged**